spec/how_is_spec.rb

```diff
@@ -14,82 +15,97 @@
   File.expand_path("./data/how-is-example-empty-repository-report.html", __dir__)
 
 JEKYLL_HEADER =
-  <<~EOF
+  <<~HEADER
     ---
     title: rubygems/rubygems report
     layout: default
     ---
-  EOF
+  HEADER
 
 describe HowIs do
-  before do
-    # 2016-11-01 00:00:00 UTC.
-    # See note in lib/how_is/report.rb about new_offset.
-    # TODO: Stop pretending to always be in UTC.
-    date = DateTime.parse("2016-11-01").new_offset(0)
-    Timecop.freeze(date)
-  end
-
-  after do
-    Timecop.return
-  end
-
   it "from_json(json) works" do
     expected = File.open(HOW_IS_EXAMPLE_REPOSITORY_JSON_REPORT).read
     actual = HowIs.from_json(expected).to_json
 
     expect(expected.strip).to eq(actual.strip)
   end
 
-  context "with a config" do
+  context "#from_config" do
+    let(:config) {
+      YAML.load_file(HOW_IS_CONFIG_FILE)
+    }
+
     it "generates valid report files" do
       Dir.mktmpdir { |dir|
         Dir.chdir(dir) {
           reports = nil
 
           VCR.use_cassette("how-is-with-config-file") do
             expect {
-              reports = HowIs.from_config(YAML.load_file(HOW_IS_CONFIG_FILE))
+              reports = HowIs.from_config(config, "2017-08-01")
             }.to_not output.to_stderr
           end
 
-          html_report = reports["./report.html"]
-          # TODO: Verify that JSON report is correct.
-          # json_report = reports['./report.json']
+          html_report = reports["output/report.html"]
+          json_report = reports["output/report.json"]
 
           expect(html_report).to include(JEKYLL_HEADER)
+          expect {
+            JSON.parse(json_report)
+          }.to_not raise_error
         }
       }
     end
-  end
 
-  context "HTML report for how-is/example-repository" do
-    it "generates a valid report" do
-      expected = File.open(HOW_IS_EXAMPLE_REPOSITORY_HTML_REPORT).read.chomp
-      actual = nil
+    it "adds correct frontmatter" do
+      reports = nil
 
-      VCR.use_cassette("how-is-example-repository") do
-        expect {
-          actual = HowIs.new("how-is/example-repository").to_html
-        }.to_not output.to_stderr
+      VCR.use_cassette("how-is-from-config-frontmatter") do
+        reports = HowIs.from_config(config, "2017-08-01")
       end
 
-      expect(expected).to eq(actual)
+      actual_html = reports["output/report.html"]
+      # actual_json = reports["output/report.json"]
```

Comment:
Q: Is this debugging commented-out code, or a note about further development?

Reply:
It's a note about potential future development. iirc, atm it doesn't apply frontmatter to JSON, and I'm not sure whether or not it's worth adding (consistency says yes, "don't add features until you need them" says no). I'll add a note being more explicit about why that's there.